### PR TITLE
Add support for contexts in interaction services dropdown

### PR DESCRIPTION
**Pull request by ztolley**
- [x] Add support for filtering metadata by context
- [x] Refactor interaction creation and edit flow to simplify and improve efficiency
- [x] Only select service options related to interaction form being edited
- [x] Improve test coverage and approach
- [x] Update acceptance tests with new service options

### CODE REVIEW

File: src/apps/interactions/services/interaction-options.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const { get, includes, snakeCase } = require('lodash')
+
+const { getContactsForCompany } = require('../../contacts/repos')
+const { getAdvisers } = require('../../adviser/repos')
+const { filterActiveAdvisers } = require('../../adviser/filters')
+const { getActiveEvents } = require('../../events/repos')
+const { transformObjectToOption, transformContactToOption } = require('../../transformers')
+const { getOptions } = require('../../../lib/options')
+
+const SERVICE_DELIVERY_STATUS_COMPLETED = '47329c18-6095-e211-a939-e4115bead28a'
```
Comment (yozzo) [inline src/apps/interactions/services/interaction-options.js]: please refer to this string from the apps constants file

File: test/unit/apps/interactions/controllers/edit.interaction.test.js [hunk 2]
```diff
@@ -0,0 +1,652 @@
+const moment = require('moment')
```
Comment (yozzo) [inline test/unit/apps/interactions/controllers/edit.interaction.test.js]: this commit seems to be a major refactor to the interaction tests. Can we have a knowledge sharing session where you can guide us through the new BDD style of writing tests?

File: test/unit/apps/interactions/controllers/edit.interaction.test.js [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+const { find, pick } = require('lodash')
+
+const config = require('~/config')
+const interactionData = require('../../../data/interactions/interaction.json')
```
Comment (reupen) [inline test/unit/apps/interactions/controllers/edit.interaction.test.js]: Why do other imports use `~` but not this one?

File: test/unit/apps/interactions/controllers/edit.service.test.js [hunk 4]
```diff
@@ -0,0 +1,593 @@
+const moment = require('moment')
+const { find, pick } = require('lodash')
+
+const config = require('~/config')
+const serviceDeliveryData = require('../../../data/interactions/service-delivery.json')
```
Comment (reupen) [inline test/unit/apps/interactions/controllers/edit.service.test.js]: Same question about `~`.

File: test/unit/apps/interactions/controllers/edit.interaction.test.js (hunk 2 above)
Comment (ztolley) [reply test/unit/apps/interactions/controllers/edit.interaction.test.js]: We had a mix of styles of testing and more recently followed BDD but ended up duplicating tests. Next time I'm in I'll share why I did this

File: src/lib/options.js [hunk 5]
```diff
@@ -24,6 +24,12 @@ async function getOptions (token, key, { createdOn, currentValue, includeDisable
     })
   }
 
+  if (context) {
```
[the remark this answers is not in the document]
Comment (ztolley) [reply src/lib/options.js]: ooo lovely, I will do that :)

File: src/apps/interactions/services/interaction-options.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    options.channels = await getOptions(token, 'communication-channel', { createdOn })
+  }
+
+  return options
+}
+
+async function getServiceOptions (req, res, createdOn) {
+  let context
+
+  if (get(req, 'params.investmentId') || get(res.locals, 'interaction.investment_project.id')) {
```
[the remark this answers is not in the document]
Comment (ztolley) [reply src/apps/interactions/services/interaction-options.js]: Done one thing and do it well, ok.

File: test/unit/apps/interactions/controllers/edit.interaction.test.js (hunk 3 above)
Comment (ztolley) [reply test/unit/apps/interactions/controllers/edit.interaction.test.js]: That import was there from a cut and paste job. I'll change it to be consistent

File: src/lib/options.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ztolley) [reply src/lib/options.js]: Could that also be used to filter out disabled options, so you would pass it sometihing like `?disable_before=20180101`?

[the remark this answers is not in the document]
Comment (reupen) [reply src/lib/options.js]: No, there are no disabled_on filters there at the moment unfortunately.

**Pull request merged**